djstripe/models/billing.py

```diff
@@ -1322,15 +1392,50 @@ class Subscription(StripeModel):
         help_text="The quantity applied to this subscription. This value will be "
         "`null` for multi-plan subscriptions",
     )
-    start = StripeDateTimeField(
+    # TODO: schedule (implement model SubscriptionSchedule, see #899)
+    legacy_start = StripeDateTimeField(
```

Comment:
`start` and `start_date` are slightly different:

From https://stripe.com/docs/upgrades#2019-10-17:

> The start field on a subscription resource has been removed and is replaced by a start_date field which represents when the entire subscription started as opposed to when the current plan configuration started.

`start` description was: 

> Date of the last substantial change to this subscription. For example, a change to the items array, or a change of status, will reset this timestamp.

while `start_date` description is:

> Date when the subscription was first created. The date might differ from the created date due to backdating

Reply:
One question here, how do you usually deal with new fields that should not be `null=True` nor `blank=True`? In this PR I made `start_date` nullable which doesn't feel alright.

When applying this migration, should we populate this field either with the previous `start` value or syncing from the API? What is the preferred way?

Reply:
> 
> 
> `start` and `start_date` are slightly different:
> 
> From https://stripe.com/docs/upgrades#2019-10-17:

OK fair enough - we should probably also upgrade to api version >= 2019-10-17 as part of 2.3, though that can be a separate change (I've added #1101 for that).

Reply:
> One question here, how do you usually deal with new fields that should not be `null=True` nor `blank=True`? In this PR I made `start_date` nullable which doesn't feel alright.
>
> When applying this migration, should we populate this field either with the previous `start` value or syncing from the API? What is the preferred way?

Yeah for new fields on existing models I'd normally set them `null=True` + `blank=True` - we don't have a safe non-null value, users can then back-fill by running `./manage.py djstripe_sync_models` when they want.

I wouldn't try and guess what the values should be, and I wouldn't want us to do this sync automatically in the migration since it might take a long time.

Reply:
:+1: